sos/plugins/openstack_glance.py

```diff
@@ -44,6 +46,18 @@ def setup(self):
 
         self.add_copy_spec("/etc/glance/")
 
+        run_cmds = False
+        for os_var in ['OS_USERNAME', 'OS_PASSWORD', 'OS_TENANT_NAME']:
+            if os_var not in os.environ:
+                self.soslog.warning("%s not found in environment variables"
+                                    " which is required to run commands"
+                                    % (os_var))
+            else:
+                run_cmds = True
```

Comment:
Do you want to set run_cmds to True if and only if *all* the three variables are set? If so, then this wont work.. Check os.environ containing just one of the three vars - you get two warnings but run_cmds=True.

(or is your intent to set run_cmds if at least one such variable is set? that's what the code does..)